src/store/eventing/middleware.ts

```diff
@@ -11,17 +11,21 @@ import { handleActiveAnnotationEvents } from './active';
 import { handleAnnotationsInitialized } from './init';
 import { handleCreateErrorEvents, handleCreatePendingEvents, handleCreateSuccessEvents } from './create';
 import { handleFetchErrorEvents } from './fetch';
+import { handleResetCreatorAction, handleSetStagedAction } from './staged';
 import { handleToggleAnnotationModeAction } from './mode';
 import { toggleAnnotationModeAction } from '../common/actions';
+import { setStagedAction, resetCreatorAction } from '../creator';
 
 // Array of event handlers based on redux action. To add handling for new events add an entry keyed by action
 const eventHandlers: EventHandlerMap = {
     [createAnnotationAction.fulfilled.toString()]: handleCreateSuccessEvents,
     [createAnnotationAction.pending.toString()]: handleCreatePendingEvents,
     [createAnnotationAction.rejected.toString()]: handleCreateErrorEvents,
     [fetchAnnotationsAction.rejected.toString()]: handleFetchErrorEvents,
+    [resetCreatorAction.toString()]: handleResetCreatorAction,
     [setActiveAnnotationIdAction.toString()]: handleActiveAnnotationEvents,
     [setIsInitialized.toString()]: handleAnnotationsInitialized,
+    [setStagedAction.toString()]: handleSetStagedAction,
```

Comment:
Do we need to handle `setStatusAction`? Or can it even take the place of `setStagedAction` here?

Reply:
I think we should stick to `setStagedAction` since the transitions in state for `staged` allows us to infer the event status as well as the annotation type